### PR TITLE
new convergence condition: only executor

### PR DESCRIPTION
**Pull request by mohamadyounes33**
- We have convergence of each worker when train loss increased and valid loss decrease at some point => prevent overfitting.

### CODE REVIEW

File: diagnosenet/executors.py [hunk 1]
```diff
@@ -6,13 +6,13 @@
...
 import time
-from typing import Sequence, NamedTuple
+from typing import NamedTuple
 
 import numpy as np
 import tensorflow as tf
 from sklearn.metrics import f1_score
 
-from diagnosenet.datamanager import Dataset, Batching, MultiTask
+from diagnosenet.datamanager import Dataset, Batching
```
Comment (jagh) [inline diagnosenet/executors.py:15]: Why do you delete the class Multitask and the Typing Sequence?

File: diagnosenet/executors.py [hunk 2]
```diff
@@ -1367,11 +1367,12 @@ def training_multigpu(self, inputs: np.ndarray, targets: np.ndarray) -> tf.Tenso
 class Distibuted_MPI:
 
     def __init__(self, model, monitor: enerGyPU = None, datamanager: Dataset = None,
-                    max_epochs: int = 10, min_loss: float = 2.0, early_stopping: int = 5) -> None:
+                 max_epochs: int = 10, min_loss: float = float("Infinity"), early_stopping: int = 5) -> None:
```
Comment (jagh) [inline diagnosenet/executors.py]: Due that we are going to initialize `min_loss` and `min_train_loss` as infinity number, doesn't matter to keep `min_loss` as a class parameter. Instead of it, we could replace it as `early_stopping= True` flag. In which the user could turn off or left on.

File: diagnosenet/executors.py [hunk 3]
```diff
@@ -1367,11 +1367,12 @@ def training_multigpu(self, inputs: np.ndarray, targets: np.ndarray) -> tf.Tenso
...
         self.model = model
         self.data = datamanager
         self.max_epochs = max_epochs
         self.min_loss = min_loss
```
Comment (jagh) [inline diagnosenet/executors.py]: We could rename `min_loss` as `min_valid_loss`

File: diagnosenet/executors.py [hunk 4]
```diff
@@ -1601,19 +1601,14 @@ def asynchronous_training(self, dataset_name: str, dataset_path: str,
                 self.training_track.append((epoch, loss, val_loss, acc, val_acc, np.round(epoch_elapsed, decimals=4)))
```
Comment (jagh) [inline diagnosenet/executors.py]: Could you rename `loss` ad `train_loss` and `val_loss` as a `valid_loss`

**Review by jagh — CHANGES_REQUESTED** [review]

File: diagnosenet/executors.py (hunk 1 above)
Comment (mohamadyounes33) [reply diagnosenet/executors.py:15]: I didn't delete it. maybe because you added it later and it wasn't in rosie-mpi branch.

File: diagnosenet/executors.py (hunk 3 above)
Comment (mohamadyounes33) [reply diagnosenet/executors.py]: understood

File: diagnosenet/executors.py (hunk 2 above)
Comment (mohamadyounes33) [reply diagnosenet/executors.py]: Got it 👌

File: diagnosenet/executors.py (hunk 4 above)
Comment (mohamadyounes33) [reply diagnosenet/executors.py]: 👍

File: diagnosenet/executors.py (hunk 1 above)
Comment (jagh) [reply diagnosenet/executors.py:15]: According, could you add the imports on the working version:
from typing import Sequence, NamedTuple
from diagnosenet.datamanager import Dataset, Batching, MultiTask

**Review by jagh — APPROVED** [review]

**Review by jagh — APPROVED** [review]
